src/java.base/share/classes/jdk/internal/foreign/CABI.java

```diff
@@ -39,21 +39,19 @@ public enum CABI {
     LINUX_RISCV_64;
 
     private static final CABI ABI;
-    private static final String ARCH;
     private static final long ADDRESS_SIZE;
 
     static {
-        ARCH = StaticProperty.osArch();
         ADDRESS_SIZE = ADDRESS.bitSize();
         // might be running in a 32-bit VM on a 64-bit platform.
         // addressSize will be correctly 32
-        if ((ARCH.equals("amd64") || ARCH.equals("x86_64")) && ADDRESS_SIZE == 64) {
+        if (Architecture.isX64() && ADDRESS_SIZE == 64) {
```

Comment:
Is there a difference to Architecture.is64bit (I.e. is the later or the former runtime vs compiletime

Reply:
There should be no difference; I was hesitant to drop the ADDRESS_SIZE check without knowing more about the foreign api dependencies.  ADDRESS_SIZE is computed (I think) from `UNSAFE.ADDRESS_SIZE * 8`.
But I can't think of how it can be different than the CPU_BITS that are defined when the JDK is built.

Reply:
> Is there a difference to Architecture.is64bit (I.e. is the later or the former runtime vs compiletime

Theoretically, the two can be unrelated. Linux x32 ABI is a typical example of using 32-bit addresses on the x86-64 architecture.

Reply:
Right, it is still possible to run a 32-bit VM build on a 64-bit platform. `os.arch` is runtime platform in this case, but the VM variant can still be 32 bits.

Since `Architecture.isX64` seems to be the target arch of the VM, I'm think that when running a 32-bit VM on a amd64 would result in `isX64` being `false`? (In that case, only the `ixX64` check should be enough)

Reply:
> Right, it is still possible to run a 32-bit VM build on a 64-bit platform. `os.arch` can be `amd64`, the runtime platform, in this case, but the VM variant can still be 32 bits.

`os.arch` is always the target architecture for JVM, even using 32-bit JVMs on 64 bit machines, its value remains x86.

But as mentioned above, the size of the address can be independent of these. 

Linux x32 ABI is an ABI for the x86-64 architecture, which uses unique features of 64 bit architectures such as 64 bit registers, but it uses 32-bit addresses.

Although OpenJDK does not support such a platform, technically speaking, it is still necessary to distinguish between architecture bits and address bits.

Reply:
> os.arch is always the target architecture for JVM, even using 32-bit JVM on 64-bit system, its value remains x86.

Okay, I could have sworn we had to add the `ADDRESS_SIZE` check because of this issue where a 32-bit VM running on x64 would erroneously select the x64 ABIs.